### PR TITLE
dra e2e: fix "reallocation works" flake

### LINKED ISSUE #119175: DRA E2E: "multiple nodes reallocation works" flake
### What happened?

**What happened**:

Shortly after https://github.com/kubernetes/kubernetes/pull/118862 was merged, the "multiple nodes reallocation works" E2E DRA test started to fail randomly.

**Please provide links to example occurrences, if any**:

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kind-dra/1677313694523985920

/sig node
/kind bug

### What did you expect to happen?

 https://testgrid.k8s.io/sig-node-dynamic-resource-allocation#ci-kind-dra needs to stay green.

### How can we reproduce it (as minimally and precisely as possible)?

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kind-dra/1677313694523985920

### Anything else we need to know?

_No response_

### Kubernetes version

master

### Cloud provider

n/a

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

The main problem probably was that
https://github.com/kubernetes/kubernetes/pull/118862 moved creating the first pod before setting up the callback which blocks allocating one claim for that pod. This is racy because allocations happen in the background.

#### Which issue(s) this PR fixes:
Fixes #119175

#### Special notes for your reviewer:

The test also was unnecessarily complex and hard to read:
- The intended effect can be achieved with three instead of four claims.
- It wasn't clear which claim has "external-claim-other" as name. Using the claim variable avoids that.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/dra/dra.go [hunk 1]
```diff
@@ -476,51 +476,44 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 
 				parameters1 := b.parameters()
 				parameters2 := b2.parameters()
-				pod1 := b.podExternal()
-				pod2 := b2.podExternal()
+				// Order is relevant here: each pod must be matched with itself
```
Comment (byako) [inline test/e2e/dra/dra.go:479]:
```suggestion
				// Order is relevant here: each pod must be matched with its
```

**Review by byako — CHANGES_REQUESTED** [review]
So it is enough to fail allocation completely, without succeeding for some claims? Alright, it is also simpler with one claim.

File: test/e2e/dra/dra.go (hunk 1 above)
Comment (pohly) [reply test/e2e/dra/dra.go]: Fixed, thanks.

**Comment by pohly** [comment]
> So it is enough to fail allocation completely, without succeeding for some claims?

Perhaps I don't understand, but of course a driver can fail allocation completely. There's no way how it can guarantee to allocate at least one claim.

**Comment by byako** [comment]
I meant the test case, original implementation had pod with two claims for same driver, with one getting allocated, the other one failing after no more resources left. So I thought there had to be partial failure (one out of two claims failed allocation) to trigger reallocation.
Never mind.

**Pull request merged**